airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/formats/abstract_file_parser.py

```diff
@@ -63,35 +78,28 @@ def json_type_to_pyarrow_type(typ: str, reverse: bool = False, logger: AirbyteLo
         :return: PyArrow type if reverse is False, else Json type
         """
         str_typ = str(typ)
-        # this is a map of airbyte types to pyarrow types. The first list element of the pyarrow types should be the one to use where required.
-        map = {
-            "boolean": ("bool_", "bool"),
-            "integer": ("int64", "int8", "int16", "int32", "uint8", "uint16", "uint32", "uint64"),
-            "number": ("float64", "float16", "float32", "decimal128", "decimal256", "halffloat", "float", "double"),
-            "string": ("large_string", "string"),
-            # TODO: support object type rather than coercing to string
-            "object": ("large_string",),
-            # TODO: support array type rather than coercing to string
-            "array": ("large_string",),
-            "null": ("large_string",),
-        }
+        # This is a map of airbyte types to pyarrow types.
+        # The first list element of the pyarrow types should be the one to use where required.
+
         if not reverse:
-            for json_type, pyarrow_types in map.items():
+            for json_type, pyarrow_types in cls.TYPE_MAP.items():
                 if str_typ.lower() == json_type:
-                    return str(
-                        getattr(pa, pyarrow_types[0]).__call__()
-                    )  # better way might be necessary when we decide to handle more type complexity
+                    type_ = next(iter(pyarrow_types))
+                    if type_ in cls.NON_SCALAR_TYPES:
+                        return cls.NON_SCALAR_TYPES[type_]
```

Comment:
I'm not sure I understand this piece of code:
* Why are you calling `next(iter(pyarrow_types))`? 
* The `struct` type is not in any `pyarrow_types` tuple, so `if type_ in cls.NON_SCALAR_TYPES:` is never `true`. Am I right?

Reply:
Oh I saw that you overrode `TYPE_MAP` in JsonlParser. Why did you chose to not map `object` to `struct` in `AbstractFileParser` ?

Reply:
- `next(iter(pyarrow_types))` is interchangable with pyarrow_types[0] - I just dont like we're getting element by a hardcoded index. Still I can revert it as it's the matter of taste and wont affect functionality

- `pyarrow_types` is not same as `NON_SCALAR_TYPES` and the first of them does not contain `struct`, so the expression above is always `true` for `struct`

- Mapping `object` to `struct` in a base class introduces a risk of breaking something, like csv parser for instance. Current  architecture of classes is a bit fragile. Overriding `TYPE_MAP` in `JsonlParser` allows us decrease the scope of code that is affected by the introduced change